### DISCUSSION TITLE
Is an ESP32 S3 with external I2S DAC on platformio a working configuration?

### DISCUSSION THREAD

**Discussion by joba-1** [pschatzmann/arduino-audio-tools#1787] [Q&A]
Hi all,

I'd like to transfer audio from an I2S mic of one ESP32-S3 over serial/RS485 to another ESP32-S3 with I2S DAC, but I'm stuck at the very beginning: I don't get a generated sine wave to play properly on the DAC. It generates more like two overlayed sine waves. On a normal ESP32 the sound is fine.

I have an ESP32-S3-DevKit C and the max98357a dac pins bck, lrc and din are connected to pins 12, 11 and 14. I use the streams-generator-i2s.ino example and just changed AudioLogger level to warning and set the out config pins in setup().

The build mentiones these versions, in case it matters:
```
PACKAGES: 
 - framework-arduinoespressif32 @ 3.20017.0 (2.0.17) 
 - tool-esptoolpy @ 1.40501.0 (4.5.1) 
 - tool-mkfatfs @ 2.0.1 
 - tool-mklittlefs @ 1.203.210628 (2.3) 
 - tool-mkspiffs @ 2.230.0 (2.30) 
 - tool-openocd-esp32 @ 2.1100.20220706 (11.0) 
 - toolchain-riscv32-esp @ 8.4.0+2021r2-patch5 
 - toolchain-xtensa-esp32s3 @ 8.4.0+2021r2-patch5

Dependency Graph
|-- audio-tools @ 0.9.9+sha.2d5d09a
```

platformio.ini looks like this (mhetesp32minikit works using pins 21, 16, 17):
```ini
[env]
platform = espressif32 @ 6.8.1
framework = arduino
lib_deps = https://github.com/pschatzmann/arduino-audio-tools.git
monitor_filters = esp32_exception_decoder
monitor_speed = 115200


[env:esp32-s3-devkitc-1]
board = esp32-s3-devkitc-1
build_flags = -DARDUINO_USB_CDC_ON_BOOT=1 -DBOARD_S3C1

build_type = debug
debug_tool = esp-builtin
debug_load_mode = manual
debug_speed = 40000
; debug_init_break = break setup

monitor_port = /dev/ttyACM_s3_1_jtag
;monitor_port = /dev/ttyACM_s3_1_com

upload_port = /dev/ttyACM_s3_1_jtag
upload_speed = 2000000
upload_protocol = esp-builtin


[env:mhetesp32minikit]
board = mhetesp32minikit
build_flags = -DBOARD_MINI 

monitor_port = /dev/ttyACM0

upload_port = /dev/ttyACM0
```

the changed part of the setup() function

```c++
...
AudioLogger::instance().begin(Serial, AudioLogger::Warning);

  // start I2S
  Serial.println("starting I2S...");
  auto config = out.defaultConfig(TX_MODE);
  config.copyFrom(info);

  #if defined(BOARD_S3C1)

    config.pin_bck  = 12;
    config.pin_ws   = 11;
    config.pin_data = 14;

    pinMode(13, OUTPUT);    // set dac gain pin...
    digitalWrite(13, LOW);  // ...to low=+12dB or high=+6dB

  #elif defined(BOARD_MINI)

    config.pin_bck  = 21;
    config.pin_ws   = 16;
    config.pin_data = 17;
    
  #endif
  
  out.begin(config);
  ...
```
any hints what I could try next?

**Comment by joba-1** [comment]
I also did a test without the arduino-audio-tools and got the same result: https://github.com/joba-1/esp32-s3-dac-sine/blob/master/src/main.cpp

So I'm sure it is not a problem of the arduino-audio-tools. Hoping for a workaround or a known good framework version...
 
The S3 board I'm using:
![image](https://github.com/user-attachments/assets/54fc8ba4-ca09-4b31-a275-a53bd3b2f74e)

**Comment by pschatzmann** [comment]
Not sure what this could be, after all the API is the same: so it should work.
- try different pins
- maybe the power pins are too noise
- try some different sample rates
- make sure that the cables are very short (to reduce the chance of noise)

**Reply by joba-1** [reply]
Thanks!

* Dirty power is always possible, usually it is USB power from the notebook, I'll try other options.
* Sample rates 8000 and 16000 also checked, same result.
* Cables are very short, 1-4 cm (wires are flat on a breadboard, no dupont, the dac vcc and gnd is directly connected to the ESPs 5V/Gnd rails, no cable at all)
* I'll try different pins, and currently installing arduino ide to get newest espressif framework...

Good luck the coming days!

**Comment by joba-1** [comment]
same with newest arduino ide, newest esp32 package from espressif and your newest lib.

This is how it looks like:
![image](https://github.com/user-attachments/assets/c4a126a0-6dea-4fa1-8329-eda66d60381c)

hoping for pin change effects…

**Comment by joba-1** [comment]
Success!

Pins 37, 36, 35 produce a nice and clean B4 :)
